### PR TITLE
chore: refactor chat provider

### PR DESCRIPTION
**Pull request by codeincontext**
Please review commit-by-commit

I'm tracking down a bug in the UI, and as a first start I thought that it would help to see the inner workings of the chat provider more clearly

By the way: I accidentally merged and subsequently reverted an earlier version of this PR

### CODE REVIEW

File: apps/nextjs/src/components/ContextProviders/ChatProvider.tsx [hunk 1]
```diff
@@ -113,26 +113,12 @@ export function ChatProvider({ id, children }: Readonly<ChatProviderProps>) {
...
     refetch: refetchModerations,
-  } = trpc.chat.appSessions.getModerations.useQuery(
-    { id },
-    {
-      refetchOnMount: true,
-      refetchOnWindowFocus: true,
-      staleTime: 0,
-    },
-  );
+  } = trpc.chat.appSessions.getModerations.useQuery({ id });
```
Comment (stefl) [inline apps/nextjs/src/components/ContextProviders/ChatProvider.tsx:235]: The refetch on mount option I think was what fixed the issue with back navigation when returning to the chat after the download page.

File: apps/nextjs/src/components/ContextProviders/ChatProvider.tsx [hunk 2]
```diff
@@ -328,9 +327,18 @@ export function ChatProvider({ id, children }: Readonly<ChatProviderProps>) {
       invokeActionMessages(response.content);
 
       trpcUtils.chat.appSessions.getChat.invalidate({ id });
+      clearHashCache();
```
Comment (stefl) [inline apps/nextjs/src/components/ContextProviders/ChatProvider.tsx:327]: Is the onFinish handler reliable? Do we need to have similar logic in onError?

File: apps/nextjs/src/components/ContextProviders/ChatProvider.tsx (hunk 1 above)
Comment (stefl) [reply apps/nextjs/src/components/ContextProviders/ChatProvider.tsx:235]: You can't actually see this error in local development. You'll need to deploy to Vercel to see it in action, so could you check it doesn't reintroduce the problem?

**Review by stefl — APPROVED** [review]
Could you check that this doesn't reintroduce the back navigation from downloads problem?

File: apps/nextjs/src/components/ContextProviders/ChatProvider.tsx (hunk 2 above)
Comment (codeincontext) [reply apps/nextjs/src/components/ContextProviders/ChatProvider.tsx:327]: Interesting, I hadn't considered that

File: apps/nextjs/src/components/ContextProviders/ChatProvider.tsx (hunk 1 above)
Comment (codeincontext) [reply apps/nextjs/src/components/ContextProviders/ChatProvider.tsx:235]: Sorry, I thought I had replied to this!

I looked at the tanstack-query docs which said that these three properties were already the defaults, so I did remove them without checking that. I'll have a test now and will see if I can get to the bottom of it

**Review by mikeritson-oak — CHANGES_REQUESTED** [review]
I spoke to @codeincontext about this and it's waiting on other PRs and work to move along before this can progress.

**Pull request closed without merging**